src/test/java/com/github/emilshina/testcases/Data.java

```diff
@@ -0,0 +1,22 @@
+package com.github.emilshina.testcases;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Custom annotation for test data.
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.METHOD)
+public @interface Data {
+
+    String source() default "";
+
+    String dataType() default "";
+
+    Class entity() default Object.class;
+
+    Class entityArr() default Object[].class;
```

Comment:
it's not required to explicitly set parameters for single class and array... you can use only one of them... generic type is more valuable... + your data processor should be smart enough to determine if there's a single object or array in a file, and make corresponding conversions on fly...